src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -308,27 +308,42 @@ export class RunParametersAction extends TooltipFromLabelAction {
 		const editor = this._notebookService.findNotebookEditor(context);
 		// Set defaultParameters to the parameter values in parameter cell
 		let defaultParameters = new Map<string, string>();
+		let emptyParameterCell: boolean;
 		editor.cells.forEach(cell => {
 			if (cell.isParameter) {
+				// Check if cell is empty or contains just whitespace
+				const cellSource = typeof cell.source === 'string' ? [cell.source] : cell.source;
+				for (let x of cellSource) {
+					emptyParameterCell = /^\s*$/.test(x);
```

Comment:
This doesn't seem correct - if the cell source is an array then you're just checking whether ANY of the values are empty. So if you have a bunch of non-empty values and then one empty value it'll think the cell is "empty", which isn't correct. 

Instead you should probably be doing `emptyParameterCell = cellSource.every(s => /^\s*$/.test(s))`